ROX-18288: compliance metadata datastores

dashrews78 opened the pull request:
## Description

Built the datastores for compliance integration metadata.
Put a unique constraint on the cluster so that we only have one metadata object 
Added unit test.
Wiring up the data model to the sensor and to the service will come under subsequent PRs.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

UT

dashrews78 committed:
X-Smart-Branch-Parent: master

dashrews78 committed:
X-Smart-Branch-Parent: master

dashrews78 committed:
the start

md2119 commented on central/complianceoperatorintegration/datastore/datastore_impl.go:
```diff
@@ -0,0 +1,127 @@
+	var integrationID string
+
+	existingIntegrations, err := ds.storage.GetByQuery(ctx, search.NewQueryBuilder().
+		AddStrings(search.ClusterID, clusterID).ProtoQuery())
```
`AddExactMatch`

md2119 commented on central/complianceoperatorintegration/datastore/datastore_impl.go:
```diff
@@ -0,0 +1,127 @@
+		return "", errors.Wrapf(err, "Multiple integrations already registered to cluster %q", clusterID)
+	}
+
+	if len(existingIntegrations) == 1 {
+		integrationID = existingIntegrations[0].Id
+	} else {
+		integrationID = uuid.NewV4().String()
+	}
```
This pattern for add obj path is incorrect. This should not be handled in the datastore but by the pipeline manager.

dashrews78 replied:
Yeah I went back and forth on that.  I will remove this logic.  It will re-appear later when I implement the pipeline manager.

dashrews78 committed:
X-Smart-Squash: Squashed 10 commits:

b1d3de5f5f getting started
60907c9341 regen
bfcb43b69e some more work
af28b85f7a some more progress
a986bb7482 some implementation
7e3e69fe0d tests
db67de77b4 add build tag
5591aa0c41 unique on cluster
b2fbcac22d stupid test build tag
fa7c47ada1 some cleanup on datastore per comments

dashrews78 committed:
rebase of generated
